cl_dll/hud_oldscoreboard.cpp

```diff
@@ -0,0 +1,309 @@
+#include "hud.h"
+#include "cl_util.h"
+#include <string.h>
+#include <stdio.h>
+#include <demo_api.h>
+#include "vgui_TeamFortressViewport.h"
+#include "hud_oldscoreboard.h"
+#include "vgui_ScorePanel.h"
+
+// Y positions
+// Those who play on killed-off 32bit Apple don't deserve old_scoreboard looking good
+// as I am too lazy to build & test in a MacOS, sorry.
+#ifdef LINUX
+	#define ROW_GAP (gHUD.m_scrinfo.iCharHeight - 5)
+	#define ROW_RANGE_MIN (gHUD.m_scrinfo.iCharHeight + 3)
+
+	/* The scoreboard's default width is set to 380 on Linux
+	 * since it doesn't look good (Ping/loss doesn't fit and overlaps "Deaths" etc.)
+	 * on the (original) 320 default, like it does on Windows,
+	 * since the engine's Trebuchet MS is bigger than on Windows
+	 * WINDOWS = Trebuchet MS = 9x8
+	 * LINUX = Trebuchet MS = 11x10
+	 */
+	#define DEFAULT_WIDTH "380"
+	#define DEFAULT_WIDTH_NUM 380
+#else
+	#define ROW_GAP  (gHUD.m_scrinfo.iCharHeight - 5)
+	#define ROW_RANGE_MIN (gHUD.m_scrinfo.iCharHeight + 2)
+	#define DEFAULT_WIDTH "320"
+	#define DEFAULT_WIDTH_NUM 320
+#endif // LINUX
+
+#define ROW_RANGE_MAX ( ScreenHeight - 50 )
+#define ROW_TOP 40
+
+#define TEAM_NO				0
+#define TEAM_YES			1
+#define TEAM_SPECTATORS		2
+#define TEAM_BLANK			3
+
+int CHudOldScoreboard::Init(void)
+{
+	m_pCvarOldScoreboard = CVAR_CREATE("cl_old_scoreboard", "0", FCVAR_ARCHIVE);
+	m_pCvarOldScoreboardWidth = CVAR_CREATE("cl_old_scoreboard_width", DEFAULT_WIDTH, FCVAR_ARCHIVE);
+
+	m_bShowScoreboard = false;
+	m_iFlags = 0;
+
+	gHUD.AddHudElem(this);
+	return 1;
+};
+
+int CHudOldScoreboard::VidInit(void)
+{
+	m_iFlags |= HUD_ACTIVE;
+	m_iFlags |= HUD_INTERMISSION; // is always drawn during an intermission
+
+	int m_iSprite = 0;
+	m_iSprite = gHUD.GetSpriteIndex("icon_ctf_score");
+	m_IconFlagScore.spr = gHUD.GetSprite(m_iSprite);
+	m_IconFlagScore.rc = gHUD.GetSpriteRect(m_iSprite);
+	m_WidthScale = m_pCvarOldScoreboardWidth->value / 320.0f;
+	m_bShowScoreboard = false;
+
+	return 1;
+}
+
+void CHudOldScoreboard::Reset(void)
+{
+
+}
+
+bool CHudOldScoreboard::IsVisible()
+{
+	return m_bShowScoreboard;
+}
+
+void CHudOldScoreboard::ShowScoreboard(bool bShow)
+{
+	if (bShow && gViewPort && gViewPort->m_pScoreBoard)
+		gViewPort->m_pScoreBoard->RebuildTeams();
+	m_bShowScoreboard = bShow;
+}
+
+int CHudOldScoreboard::Draw(float fTime)
+{
+	// Let users use 320 even on Linux, if they really want to
+	if (m_pCvarOldScoreboardWidth->value < 320.0f || m_pCvarOldScoreboardWidth->value > ScreenWidth)
+	{
+		gEngfuncs.Con_Printf("Invalid cl_oldscoreboard_width value (%d) ", (int)m_pCvarOldScoreboardWidth->value);
+		gEngfuncs.Con_Printf("(minimum = %d, maximum = %d (current screen width)\n", 320, ScreenWidth);
+		gEngfuncs.Con_Printf("Resetting to default: %s\n", DEFAULT_WIDTH);
+		m_pCvarOldScoreboardWidth->value = (float)DEFAULT_WIDTH_NUM;
```

Comment:
Does that actually update the cvar string properly?

Reply:
Right, it only updated the "internal" value, not the value shown to user. Fixed with `gEngfuncs.Cvar_SetValue("cl_old_scoreboard_width", DEFAULT_WIDTH_NUM);`

This obviously still only changes the value after the user is back in the game as HUD Draw() functions aren't called in the menu (where you use the console).